packages/core/src/util/renderer.ts

```diff
@@ -80,11 +82,284 @@ import {
 } from '../i18n/arrayTranslations';
 import { resolveSchema } from './resolvers';
 import cloneDeep from 'lodash/cloneDeep';
+import { has } from 'lodash';
+import { all, any } from 'lodash/fp';
+
+const checkDataCondition = (
+  propertyCondition: unknown,
+  property: string,
+  data: Record<string, unknown>
+) => {
+  if (has(propertyCondition, 'const')) {
+    return (
+      has(data, property) && data[property] === get(propertyCondition, 'const')
+    );
+  } else if (has(propertyCondition, 'enum')) {
+    return (
+      has(data, property) &&
+      (get(propertyCondition, 'enum') as unknown[]).includes(data[property])
+    );
```

Comment:
These will only work when values are primitives. If a value is an object these comparisons will fail. We should probably use lodash's `isEqual` here to cover the more complex use cases too.

Reply:
You are right, there were other places as well where it was needed to compare the values with `isEqual`. It is fixed in my latest commit.